### DISCUSSION TITLE
Questions about artifactories and key-only artifacts

### DISCUSSION THREAD

**Discussion by jeffmax** [argoproj/argo-workflows#5298] [Q&A] [answered]
I have a couple questions about artifacts and artifactories:

1. What is the difference between key-only artifacts and the functionality available in Argo 2 where you can use a default artifactory repository defined in the [workflow-config configmap](https://github.com/argoproj/argo-workflows/blob/master/docs/workflow-controller-configmap.yaml#L66)? It seems like in 2, when you use the functionality described in this [ticket](https://github.com/argoproj/argo-workflows/issues/3184#issuecomment-640297735) argo will put your artifact in the the bucket without specifying any bucket or path in the actual Workflow, which seems to be what is described by a [key-only artifact](https://argoproj.github.io/argo-workflows/key-only-artifacts/), so I was wondering what the difference was. Side-by-side, it seems like the key-only artifact is specifying more in the Workflow file.

2. If you use a separate [config map for the artifactory config](https://argoproj.github.io/argo-workflows/artifact-repository-ref/), is the [keyFormat](https://github.com/argoproj/argo-workflows/blob/master/docs/workflow-controller-configmap.yaml#L88) functionality available like it is in the example for the config when it is in the main workflow controller config map?

**Comment by alexec** [comment, marked as answer]
key-only use case is for when you want to explicitly define the key, but without defining the bucket etc.

**Reply by jeffmax** [reply]
What does the `key` refer to? Is it the full path within the bucket?- How does it compare to just specifying `name`?
